test/hotspot/jtreg/runtime/Monitor/TestRecursiveLocking.java

```diff
@@ -0,0 +1,560 @@
+                    // stack. After more recursions it will grow to
+                    // [B, B ... B].
+                    assertInflated();
+                }
+            } else if (flagLockingMode == LM_MONITOR) {
+                assertInflated();
+            }
+
+            // Call runB() at the same depth as runA's depth:
+            B.runB(depth, this);
+        }
+
+        synchronized void runB(int depth, SynchronizedObject A) {
+            counter++;
+
+            if (flagLockingMode != LM_MONITOR) {
+                // Legacy tolerates endless recursions. While testing
+                // lightweight we don't go deeper than the size of the
+                // lock stack, which in this test case will be filled
+                // with a number of B-elements. See comment in runA()
+                // above for more info.
+                assertNotInflated();
+            } else {
+                assertInflated();
+            }
+
+            if (depth == 1) {
+                // Reached LockStackCapacity in depth so we're done.
+                return;
+            } else {
+                A.runA(depth - 1, this);
+            }
+        }
+
+        // This test alternates by locking A and B.
+        public void runAlternateABTest() {
+            final SynchronizedObject A = new SynchronizedObject();
+            final SynchronizedObject B = new SynchronizedObject();
+
+            // Just checking since they are new objects:
+            A.assertNotInflated();
+            B.assertNotInflated();
+
+            A.runA(constLockStackCapacity, B);
+
+            // Verify that the nested monitors have been properly released:
+            syncThread.verifyCanBeSynced(A);
+            syncThread.verifyCanBeSynced(B);
+
+            Asserts.assertEquals(A.counter, constLockStackCapacity);
+            Asserts.assertEquals(B.counter, constLockStackCapacity);
+            if (flagLockingMode == LM_LEGACY) {
+                A.assertNotInflated();
+            }
+            // Implied else: for LM_MONITOR or LM_LIGHTWEIGHT it can be
+            // either inflated or not point because A is not locked anymore
+            // and subject to deflation.
+
+            if (flagLockingMode != LM_MONITOR) {
+                B.assertNotInflated();
+            }
+        }
+
+        void assertNotInflated() {
+            Asserts.assertFalse(WB.isMonitorInflated(this));
+        }
+
+        void assertInflated() {
+            Asserts.assertTrue(WB.isMonitorInflated(this));
+        }
+    }
+
+    static void usage() {
+        System.err.println();
+        System.err.println("Usage: java TestRecursiveLocking [n_secs]");
+        System.err.println("       java TestRecursiveLocking n_secs [mode]");
+        System.err.println();
+        System.err.println("where:");
+        System.err.println("    n_secs  ::= > 0");
+        System.err.println("            Default n_secs is 15.");
```

Comment:
Based on L408 below, it looks like default n_secs is now `30`.

Reply:
fixed